File: src/FetchBannerlordVersion.Native.TypeScript/src/lib/FetchBannerlordVersion.ts
import { IFetchBannerlordVersion } from "./types";

const fetcher: IFetchBannerlordVersion & { allocAliveCount(): number; } = require('./../../fetchblversion.node');

export const allocAliveCount = (): number => {
  return fetcher.allocAliveCount();
}

export class FetchBannerlordVersion {
  /* istanbul ignore next */
  private constructor() { }

  static getChangeSet(gameFolderPath: string, libAssembly: string): number {
    return fetcher.getChangeSet(gameFolderPath, libAssembly);
  }

  static getVersion(gameFolderPath: string, libAssembly: string): string {
    return fetcher.getVersion(gameFolderPath, libAssembly);
  }

  static getVersionType(gameFolderPath: string, libAssembly: string): number {
    return fetcher.getVersionType(gameFolderPath, libAssembly);
  }
}

File: src/FetchBannerlordVersion.DependencyInjection/FetchBannerlordVersion.cs
namespace FetchBannerlordVersion.DependencyInjection;

public class FetchBannerlordVersion : IFetchBannerlordVersion
{
    public int GetChangeSet(string gameFolderPath, string libAssembly) => Fetcher.GetChangeSet(gameFolderPath, libAssembly);

    public string GetVersion(string gameFolderPath, string libAssembly) => Fetcher.GetVersion(gameFolderPath, libAssembly);

    public VersionType GetVersionType(string gameFolderPath, string libAssembly) => Fetcher.GetVersionType(gameFolderPath, libAssembly);
}